Compiler.cmake

```diff
@@ -11,6 +11,7 @@
 # Input Variables
 # * COMMON_MINIMUM_GCC_VERSION check for a minimum gcc version, default 4.4
 # * COMMON_USE_CXX03 When set, do not enable C++11 language features
+# * COMMON_DISABLE When set, do not set any flags
```

Comment:
Yeah, agreed about the problem.  CMake 3 seems to have target_compile_features and target_compile_definitions which seems like the future.

I'm generally fine w/ your flags, but Juan was hoping for something different.  I'll see if I can get him to comment